#### <font color ='000000'> 8. Consulte que es el coeficiente de correlación de Pearson. Haga una breve descripción en un Markdown

#### <font color = '000000'> COEFICIENTE DE CORRELACIÓN DE PEARSON
   
    Es un método propio de la estadística paramétrica, el cual se usa para conocer la relacion entre dos variables cuantitativas, pero también se usa para la predicción de una variable (la regresión lineal).
    
    EL coeficiente de correlación de Pearson es una prueba que mide la relación estadística entre dos variales continuas. Si la asociación entre los elementos no es lineal, entonces el coeficiente no se encuentra representado adecuadamente, es adecuado para datos de escala de razón o intervalo, ya que toma en cuenta la magnitud relativa de las observaciones.
    
    Este coeficiente oscila entre -1 y +1 de tal modo que se interpreta de la siguiente manera:
    
    - Un valor menor que 0 indica que existe una correlación negativa, es decir, que las dos variables están asociadas en sentido inverso. Cuánto más se acerca a -1, mayor es la fuerza de esa relación invertida (cuando el valor en una sea muy alto, el valor en la otra será muy bajo). Cuando es exactamente -1, eso significa que tienen una correlación negativa perfecta.
   
    - Un valor mayor que 0 indica que existe una correlación positiva. En este caso las variables estarían asociadas en sentido directo. Cuanto más cerca de +1, más alta es su asociación. Un valor exacto de +1 indicaría una relación lineal positiva perfecta.
    
    -Una correlación de 0, o próxima a 0, indica que no hay relación lineal entre las dos variables.
    
    De tal forma el coeficiente de correlación de Pearson se define:

$$r_{xy}= \frac{\sum Z_{x}Z_{y}}{N}$$
    

#### <fotn color ='000000'> donde:
    "x"  es igual a la variable número uno
    "y"  es igual a la variable número dos
    "zx" es la desviación estándar de la variable uno
    "zy" es la desviación estándar de la variable dos
    "N"  es es número de datos.

#### <font color ='000000'> 9. Calcule la correlación entre el PIB departamental y los resultados de las pruebas SABER 11 en la asignatura de matemáticas (Variable “tema_matematica”). ¿Qué puede decir sobre la relación entre estas dos variables? Ayúdese de lo consultado en el punto 8. Responda la pregunta en el Markdown.
    
    PIB departamental:

In [6]:
import pandas as pd

In [22]:
PIB = pd.read_excel('PIB Departamental.xlsx', index_col=None, header=None)

In [34]:
PIB = PIB.rename (columns = {0:"Departamento", 1:"PIB"})
PIB

Departamento     PIB
0                   Amazonas     355
1                  Antioquia   67432
2                     Arauca    4895
3                  Atlántico   20429
4               Bogotá D. C.  132129
5                    Bolívar   19232
6                     Boyacá   14094
7                     Caldas    8164
8                    Caquetá    2222
9                   Casanare    8250
10                     Cauca    7341
11                     Cesar   10034
12                     Chocó    2069
13                   Córdoba    9195
14              Cundinamarca   26079
15                   Guainía     168
16                  Guaviare     465
17                     Huila    8865
18                La Guajira    6471
19                 Magdalena    7079
20                      Meta   16164
21                    Nariño    7810
22           Norte Santander    8991
23                  Putumayo    2120
24                   Quindío    3841
25                 Risaralda    7829
26  San Andrés y Providencia     772
27                 Santander   34669
28                     Sucre    4119
29                    Tolima   11571
30                     Valle   51321
31                    Vaupés     133
32                   Vichada     339

### <font color ='000000'> 
    Resultados de las pruebas SABER 11 de la asignatura de matemáticas 

In [51]:
from ipywidgets import interact, IntSlider
from IPython.display import display
def visor(df, num_rows=10, num_columns=10, step_rows=1,
                  step_columns=1):
    @interact(last_row=IntSlider(min=min(num_rows, df.shape[0]),
                                 max=df.shape[0],
                                 step=step_rows,
                                 description='rows',
                                 readout=False,
                                 disabled=False,
                                 continuous_update=True,
                                 orientation='horizontal',
                                 slider_color='purple'),
              last_column=IntSlider(min=min(num_columns, df.shape[1]),
                                    max=df.shape[1],
                                    step=step_columns,
                                    description='columns',
                                    readout=False,
                                    disabled=False,
                                    continuous_update=True,
                                    orientation='horizontal',
                                    slider_color='purple'))
    def _freeze_header(last_row, last_column):
        display(df.iloc[max(0, last_row-num_rows):last_row,
                        max(0, last_column-num_columns):last_column])

In [29]:
Saber1 = pd.read_stata('2009-1.dta')

In [30]:
Saber2 = pd.read_stata('2009-2.dta')

In [49]:
frames = [Saber1, Saber2]
ResultadosSaber11 = pd.concat (frames, ignore_index=True)

In [76]:
Saber11 = ResultadosSaber11[["cole_depto_colegio", "tema_matematica"]]
visor(Saber11)

interactive(children=(IntSlider(value=10, description='rows', max=529706, min=10, readout=False), IntSlider(va…

In [78]:
DesviaciónTemaMatematica = ResultadosSaber11["tema_matematica"]
DesviaciónTemaMatematica.std ()

10.646038055419922

In [80]:
DesviaciónDepartamentos = PIB["PIB"]
DesviaciónDepartamentos.std ()

25619.121382605354

### <font color ='000000'> CORRELACIÓN DE PEARSON ENTRE:
    PIB departamental y los resultados de la prueba SABER 11 (variable "tema_matematica")
    
    x  = PIB departamental
    y  = tema_matematica
    Zx = 25619.121382605354
    Zy = 10.646038055419922
    N  = 529705

$$r_{xy}= \frac{\sum Z_{x}Z_{y}}{N}$$

$$r_{xy}= \frac{\sum (25619.121382605354) (10.646038055419922)}{529705}$$

### <font color ='000000'> 
    Sumatoria de las desviaciones estandar

$\sum (25619.121382605354) (10.646038055419922)$

In [103]:
xy = 25619.121382605354 + 10.646038055419922
xy

25629.767420660773

### <font color ='000000'> 
   Correlación de Pearson (rxy)

$\frac{\sum (25619.121382605354) (10.646038055419922)}{529705}$

In [104]:
rxy = 25629.767420660773/529705
rxy

0.04838498300121912

$$r_{xy}= \frac{\sum Z_{x}Z_{y}}{N}$$

$$r_{xy}= 0.04838498300121912$$

### <font color ='000000'>  ¿Qué puede decir sobre la relación entre estas dos variables?
    Como el valor de la correlación es un valor mayor que 0 indica que existe una correlación positiva, es decir, que estan asociadas en sentido directo y cuanto más cerca de +1, más alta es su asociación.